src/Database.h

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2021 Chiu Yen-Chen, Swen Sun-Yen, Wen Yong-Wei.
+All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. See the AUTHORS file for names of contributors.
+ */
+
+#ifndef _DATABASE_H_
+#define _DATABASE_H_
+
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+#include "SURLCreator.h"
+
+namespace shortlink 
+{
+	// Database save the look-up table to link each SURL to there original URL
+	// public implement: Create, Read, Insert
+	// Create:		// if rawURL already exist, do UpdateSURL, else InsertSURL
+	//		string rawURL = "www.google.com"
+	//		string shortURL = "";
+	//		string createStatus = database.CreateSURL(rawURL);  
+	//		if (createStatus != "")
+	//			shortURL = createStatus;
+	//	Read:	
+	//		string rawURL = "";
+	//		string findStatus = database.ReadSURL(shortURL);	 
+	//		if (findStatus != "")
+	//			rawURL = findStatus;
+	
+	class Database
+	{
+	public:
+		Database();
+
+		// Input a rawURL, return SURL if create is successful. Return "" if create fail.
+		// Encode the rawURL to SURL and check whether it exists.
+		// If it exists update expire time, else create the URL.
+		std::string CreateSURL(const std::string& rawURL);
+
+		// Read the lut whether URL exists. Return rawURL if SURL exists. Return "" if SURL doesn't exist.
+		std::string ReadSURL(const std::string& surl);
+
+		// lock the mutex when it is inserting.
+		// There is only one inserting in the same time.
+		bool InsertSURL(std::unique_ptr<SURL>& surl);
```

Comment:
Comment that caller passes the ownership of SURL into Database.
Do not use reference for unique_ptr.